sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_path_client.py

```diff
@@ -392,6 +392,156 @@ def get_access_control(self, upn=None,  # type: Optional[bool]
         except StorageErrorException as error:
             process_storage_error(error)
 
+    @staticmethod
+    def _set_access_control_recursive_options(mode, acl=None, **kwargs):
+        # type: (Optional[ContentSettings], Optional[Dict[str, str]], **Any) -> Dict[str, Any]
+
+        options = {
+            'mode': mode,
+            'timeout': kwargs.pop('timeout', None),
+            'continuation': kwargs.pop('continuation', None),
+            'max_records': kwargs.pop('batch_size', None),
+            'acl': acl,
+            'cls': return_headers_and_deserialized}
+        options.update(kwargs)
+        return options
+
+    def set_access_control_recursive(self,
+                                     acl=None,  # type: Optional[str]
+                                     progress_callback=None,
+                                     **kwargs):
+        # type: (...) -> Dict[str, Union[str, datetime]]
+        """
+        Sets the Access Control on a path and sub-paths.
+
+        :param acl:
+            Sets POSIX access control rights on files and directories.
+            The value is a comma-separated list of access control entries. Each
+            access control entry (ACE) consists of a scope, a type, a user or
+            group identifier, and permissions in the format
+            "[scope:][type]:[id]:[permissions]".
+        :type acl: str
+        :param progress_callback:
+            Callback where the caller can track progress of the operation
+            as well as collect paths that failed to change Access Control.
+        :type progress_callback: func(~azure.storage.filedatalake._generated.models.SetAccessControlRecursiveResponse)
```

Comment:
`progress_callback` should use `keyword` and `vartype` tags

Reply:
Additionally `progress_callback` should probably not take the raw generated response - as this is not a public object in the SDK. Users will not be able to navigate to the documentation of this object.

Reply:
What if we expose this in __init__.py file?

---
In reply to: [402985432](https://github.com/Azure/azure-sdk-for-python/pull/10626#discussion_r402985432) [](ancestors = 402985432)